Framework/Geometry/src/Objects/ShapeFactory.cpp

```diff
@@ -614,32 +766,61 @@ CuboidCorners ShapeFactory::parseCuboid(Poco::XML::Element *pElem) {
     const double deltaW = getDoubleAttribute(pElem_width, "val") / 2;
     const double deltaD = getDoubleAttribute(pElem_depth, "val") / 2;
 
-    const V3D centre = pElem_centre ? parsePosition(pElem_centre) : DEFAULT_CENTRE;
+    V3D centre = pElem_centre ? parsePosition(pElem_centre) : DEFAULT_CENTRE;
 
     result.lfb = V3D(-deltaW, -deltaH, -deltaD);
     result.lft = V3D(-deltaW, deltaH, -deltaD);
     result.lbb = V3D(-deltaW, -deltaH, deltaD);
     result.rfb = V3D(deltaW, -deltaH, -deltaD);
 
     if (pElem_axis) {
-      // Use a quarternion to do a rotation for us, with respect to the default
+      // Use a quarternion to do a rotate for us, with respect to the default
       // axis.  Our "Quat" implementation requires that the vectors passed to
       // it be normalised.
       const V3D axis = normalize(parsePosition(pElem_axis));
-      const Quat rotation(DEFAULT_AXIS, axis);
+      const Quat rotate(DEFAULT_AXIS, axis);
 
-      rotation.rotate(result.lfb);
-      rotation.rotate(result.lft);
-      rotation.rotate(result.lbb);
-      rotation.rotate(result.rfb);
+      rotate.rotate(result.lfb);
+      rotate.rotate(result.lft);
+      rotate.rotate(result.lbb);
+      rotate.rotate(result.rfb);
+    }
+
+    // Rotate the points by the rotate, rotate-all and goniometer tags
+    // Rotate the centre by the rotate-all and goniometer tags
+    if (pElem_rot) {
+      // Apply manual rotation supplied to rotate tag
+      std::vector<double> rotateAngles = DegreesToRadians(parsePosition(pElem_rot));
+      const std::vector<double> rotateMatrix = generateMatrix(rotateAngles[0], rotateAngles[1], rotateAngles[2]);
+      result.lfb.rotate(rotateMatrix);
+      result.lft.rotate(rotateMatrix);
+      result.lbb.rotate(rotateMatrix);
+      result.rfb.rotate(rotateMatrix);
+    }
+    if (m_rotateAllMatrix.size() > 0) {
+      // Apply automatic rotation due to the rotate-all tag
+      result.lfb.rotate(m_rotateAllMatrix);
+      result.lft.rotate(m_rotateAllMatrix);
+      result.lbb.rotate(m_rotateAllMatrix);
+      result.rfb.rotate(m_rotateAllMatrix);
+      centre.rotate(m_rotateAllMatrix);
+    }
+    if (m_gonioRotateMatrix.size() > 0) {
+      // Apply automatic rotation due to the goniometer that should NOT be manually defined by the user
+      result.lfb.rotate(m_gonioRotateMatrix);
+      result.lft.rotate(m_gonioRotateMatrix);
+      result.lbb.rotate(m_gonioRotateMatrix);
+      result.rfb.rotate(m_gonioRotateMatrix);
```

Comment:
Similar comment to the one I've made elsewhere about the Hexahedron. This code to rotate the four corners could perhaps benefit from being put into a method on the CuboidCorners struct